src/marqo/tensor_search/tensor_search.py

```diff
@@ -1517,56 +1525,68 @@ def get_query_vectors_from_jobs(
     """
```

Comment:
the logic of `get_query_vectors_from_jobs` changes no insignificantly to allow `q.q==None`. I think there should be test cases for this scenario